### PR TITLE
Added a newsletter subscription block to the footer

### PR DESCRIPTION
**Pull request by zans-laksa**

### CODE REVIEW

**Comment by riha112** [comment]
Core review passed

File: src/app/component/NewsletterSubscription/images/letter-icon.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="18px" height="14px" viewBox="0 0 18 14" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>3946694A-9211-4FC1-AAA0-52ED9CD30EF4</title>
+    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <g id="budo&amp;fitness---footer-mobile" transform="translate(-271.000000, -610.000000)" fill="#FFFFFF" fill-rule="nonzero">
```
Comment (alfredsgenkins) [inline src/app/component/NewsletterSubscription/images/letter-icon.svg:5]: I do not think budo & fitness icon is alright here.

Comment (alfredsgenkins) [reply src/app/component/NewsletterSubscription/images/letter-icon.svg:5]: This SVG is not optimized.

Comment (alfredsgenkins) [reply src/app/component/NewsletterSubscription/images/letter-icon.svg:5]: I would appreciate if you would make this icon using CSS. Like we did other icons in `_icons.scss`.

Comment (alfredsgenkins) [reply src/app/component/NewsletterSubscription/images/letter-icon.svg:5]: This icon is pretty simple to be done in CSS

File: src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+.NewsletterSubscription {
+    --newsletter-subscription-input-background: var(--input-background-color);
+    --newsletter-subscription-input-color: #{$black};
+    --newsletter-subscription-placeholder-color: #9f9c9c;
+    --newsletter-subscription-button-background: var(--link-color);
```
Comment (alfredsgenkins) [inline src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss:16]: Please put this into the `:root`

File: src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+.NewsletterSubscription {
+    --newsletter-subscription-input-background: var(--input-background-color);
+    --newsletter-subscription-input-color: #{$black};
+    --newsletter-subscription-placeholder-color: #9f9c9c;
```
Comment (alfredsgenkins) [inline src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss:15]: What is this magic color? Where did it came from, use application-wise color pallet please!

File: src/app/component/NewsletterSubscription/NewsletterSubscription.container.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        onFormSubmit: this.onFormSubmit.bind(this)
+    };
+
+    state = {
+        isLoading: false
+    };
+
+    onFormSubmitSuccess = this.onFormSubmitSuccess.bind(this);
+
+    onFormSubmitError = this.onFormSubmitError.bind(this);
```
Comment (alfredsgenkins) [inline src/app/component/NewsletterSubscription/NewsletterSubscription.container.js:49]: You are technically doing the same thing here, bind all of this to a single function.

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]

**Comment by alfredsgenkins** [comment]
Overall, good code tho!

File: src/app/component/NewsletterSubscription/images/letter-icon.svg (hunk 1 above)
Comment (zans-laksa) [reply src/app/component/NewsletterSubscription/images/letter-icon.svg]: Done

File: src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss (hunk 2 above)
Comment (zans-laksa) [reply src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss]: Done

File: src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss (hunk 3 above)
Comment (zans-laksa) [reply src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss]: Done

File: src/app/component/NewsletterSubscription/NewsletterSubscription.container.js (hunk 4 above)
Comment (zans-laksa) [reply src/app/component/NewsletterSubscription/NewsletterSubscription.container.js]: Done

File: src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss [hunk 5]
```diff
@@ -9,12 +9,14 @@
  * @link https://github.com/scandipwa/base-theme
  */
 
-.NewsletterSubscription {
+:root {
     --newsletter-subscription-input-background: var(--input-background-color);
     --newsletter-subscription-input-color: #{$black};
-    --newsletter-subscription-placeholder-color: #9f9c9c;
+    --newsletter-subscription-placeholder-color: #{$default-secondary-dark-color};
```
Comment (alfredsgenkins) [inline src/app/component/NewsletterSubscription/NewsletterSubscription.style.scss:15]: next time use `var(secondary-dark-color)` then the varaible can be changed in the browser via color customization feature.

**Pull request merged**